src/event/EventTarget.ts

```diff
@@ -42,8 +42,12 @@ export default abstract class EventTarget {
 		let returnValue = true;
 
 		if (this._listeners[event.type]) {
+			const newTarget = Object.create(event)
```

Comment:
It does not feel good to manipulate the target property in the event object here, but that seem to be what the browser is doing. In order to simulate what the browser is doing we probably need to do the same thing here.

Suggestion:
event.target = this;